scripts/apps/users/directives/UserPreferencesDirective.ts

```diff
@@ -56,7 +54,53 @@ export function UserPreferencesDirective(
              */
 
             scope.preferencesLoaded = false;
-            var orig; // original preferences, before any changes
+            var orig: {[key: string]: any}; // original preferences, before any changes
+
+            scope.extensionsNotifications = {};
```

Comment:
What about calling this `emailNotificationsFromExtensions`?. I saw there are other kinds of notifications too. We could say that there are multiple notification types supported in extensions, but since you are interating this object in other methods and don't use any conditionals, then it makes sense to limit this object only to email notifications. If we decide in the future to modify it, it will be clear this is only used for email type and it would be easier to refactor thanks to self-documenting naming.

Reply:
I had it that way, just renamed it with its current name so it's shorter, I use it in lots of places and thought it's a good compromise